src/main/java/app/GameApplication.java

```diff
@@ -6,18 +6,28 @@
 import use_case.search.SearchInteractor;
 import view.MainFrame;
 
+/**
+ * The GameApplication class is the entry point of the application.
+ * It initializes and starts the main components of the application.
+ */
 public class GameApplication {
+
+    /**
+     * The main method to run the application.
+     *
+     * @param args the command-line arguments
+     */
     public static void main(String[] args) {
         // Step 1: Initialize dependencies for SearchInteractor
-        DBSearchDataAccessObject searchDataAccess = new DBSearchDataAccessObject();
-        SearchViewModel searchViewModel = new SearchViewModel();
-        SearchInteractor searchInteractor = new SearchInteractor(searchDataAccess, searchViewModel);
+        final DBSearchDataAccessObject searchDataAccess = new DBSearchDataAccessObject();
+        final SearchViewModel searchViewModel = new SearchViewModel();
+        final SearchInteractor searchInteractor = new SearchInteractor(searchDataAccess, searchViewModel);
 
         // Step 2: Create PlayGameController
-        PlayGameController playGameController = new PlayGameUseCaseFactory().createController();
+        final PlayGameController playGameController = new PlayGameUseCaseFactory().createController();
 
         // Step 3: Initialize MainFrame with both dependencies
-        MainFrame mainFrame = new MainFrame(playGameController, searchInteractor);
+        final MainFrame mainFrame = new MainFrame(playGameController, searchInteractor);
 
```

Comment:
Handle potential exceptions during initialization to avoid crashing the main application.